lib/srv/alpnproxy/local_proxy.go

```diff
@@ -149,7 +150,11 @@ func (l *LocalProxy) Start(ctx context.Context) error {
 
 		if l.cfg.Middleware != nil {
 			if err := l.cfg.Middleware.OnNewConnection(ctx, l, conn); err != nil {
-				log.WithError(err).Errorf("Middleware failed to handle new connection.")
+				log.WithError(err).Error("Middleware failed to handle client connection.")
+				closeErr := conn.Close()
+				if closeErr != nil {
+					log.WithError(closeErr).Debug("Failed to close client connection.")
+				}
```

Comment:
I'm fine with that, but double-closing is the only error I'm aware of for `conn.Close()` - we can filter the log like this just in case some other error occurs or just ignore the error entirely